### ISSUE TITLE
508-defect-2 [COGNITION, SCREEN READER]: Edit buttons SHOULD have accessible labels when editing inline data on Review screen

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#17824]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I want to hear an accessible label like "Edit issues eligible for review" instead of "Edit" when the button has focus or virtual cursor.

## Details

The edit button for contact information and issues eligible for review do not have descriptive aria-labels.

## Acceptance Criteria

- [ ] All edit buttons have unique and descriptive aria-labels

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Turn on a screen reader of your choice.
2. Enter `https://staging.va.gov/decision-reviews/higher-level-review/request-higher-level-review-form-20-0996/review-and-submit` in browser.
3. Complete test case 3 and reach the review page.
4. Tab to "Edit" for contact information and issues eligible for review.
5. Verify you only hear the visible label.

## Solution 

Add a more descriptive aria-label to each of these items:
- Edit contact information on profile
- Edit issues eligible for review

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

<img width="935" alt="Screen Shot 2020-12-22 at 11 52 14 AM" src="https://user-images.githubusercontent.com/14154792/102913810-4d00ca00-444d-11eb-9bac-d34503ef8b86.png">

**Comment by Mottie** [comment]
The "Edit" button inside the accordion is controlled by code within the platform form system.

I can add an `aria-label` to the "Contact information" link, but making the suggested change causes an axe coconut error "Elements must have their visible text as part of their accessible name"

![Screen Shot 2021-01-04 at 4 44 02 PM](https://user-images.githubusercontent.com/136959/103587357-b69cc180-4eac-11eb-8040-737310840785.png)

**Comment by joshkimux** [comment]
@Mottie I think we can ignore the axe coconut error in this instance because "Edit on profile" is still a part of the "Edit contact info on the profile page." Alternatively, we could use `aria-describedby` using the "Contact Information" title to describe "Edit on Profile" which creates a similar experience.

**Comment by Mottie** [comment]
I've added the recommended `aria-label` to the "Edit on profile" link in https://github.com/department-of-veterans-affairs/vets-website/pull/15550

**Comment by joshkimux** [comment]
Confirmed this has been updated. Great stuff @Mottie ! 👍

**Comment by Mottie** [comment]
Sorry, the `aria-label` inside the "Issues eligible for review" is controlled by the form system, but I forgot there is a `ui:option` that allows changing the `aria-label` - done in https://github.com/department-of-veterans-affairs/vets-website/pull/15564

**Comment by joshkimux** [comment]
@Mottie I went through staging one more time and noticed the office of review edit button does not have an `aria-label`. Temporarily re-opening this ticket, but hopefully this is an easy fix 🙇 

<img width="1393" alt="Screen Shot 2021-01-11 at 10 22 43 AM" src="https://user-images.githubusercontent.com/14154792/104201259-8c408c00-53f7-11eb-85dd-2e5ea45e0f07.png">

**Comment by Mottie** [comment]
Thanks Josh! This fix should be available in staging soon!

**Comment by joshkimux** [comment]
Confirmed this was fixed. Thank you @Mottie !!!
